backend/chainlit/server.py

```diff
@@ -909,19 +918,20 @@ async def get_avatar(avatar_id: str):
         return await get_favicon()
 
 
-@app.head("/")
+@router.head("/")
 def status_check():
     return {"message": "Site is operational"}
 
 
-def register_wildcard_route_handler():
-    @app.get("/{path:path}")
-    async def serve(request: Request, path: str):
-        html_template = get_html_template()
-        """Serve the UI files."""
-        response = HTMLResponse(content=html_template, status_code=200)
+@router.get("/{full_path:path}")
+async def serve():
+    html_template = get_html_template()
+    """Serve the UI files."""
+    response = HTMLResponse(content=html_template, status_code=200)
+
+    return response
```

Comment:
There is some convenience enabled by this function.

For example, right now I can monkey patch the `register_wildcard_route_handler` to add:
 - basic auth for a staging environment
 - header to prevent indexing by search engines

```python
...

@cl_server.app.get("/health")
def health(request: Request):
    return JSONResponse({"status": "ok"})


def wildcard_route_handler():
    security_dependency = Depends(HTTPBasic())

    @cl_server.app.get("/{path:path}")
    async def serve(
        request: Request,
        path: str,
        credentials: HTTPBasicCredentials = security_dependency,
    ):
        current_username_bytes = credentials.username.encode("utf8")
        correct_username_bytes = b"<username>"
        is_correct_username = secrets.compare_digest(
            current_username_bytes, correct_username_bytes
        )
        current_password_bytes = credentials.password.encode("utf8")
        correct_password_bytes = b"<password>"
        is_correct_password = secrets.compare_digest(
            current_password_bytes, correct_password_bytes
        )
        if not (is_correct_username and is_correct_password):
            raise HTTPException(
                status_code=status.HTTP_401_UNAUTHORIZED,
                detail="Incorrect username or password",
                headers={"WWW-Authenticate": "Basic"},
            )

        html_template = cl_server.get_html_template()
        """Serve the UI files."""
        response = HTMLResponse(content=html_template, status_code=200)
        response.headers["X-Robots-Tag"] = "noindex"
        return response


cl_server.register_wildcard_route_handler = wildcard_route_handler


from chainlit.cli import run_chainlit  # noqa: E402

run_chainlit("nectar_ui.py")
```

Hopefully a clean way to do these modifications is preserved.

Reply:
I realized I can probably write fastapi middleware that will accomplish the same thing!

Reply:
Or mount the chainlit app on your main fast api app, way cleaner!